shard/src/api/api.py

```diff
@@ -73,8 +77,47 @@ async def set_response(self, user_id, guild_id, trigger, response):
 
     @fetch_guild
     async def get_playlist(self, guild):
-        voice = self.bot.cogs['Voice'].voice_managers[guild.id]
-        return voice.q.as_dict(), sc.OK_200
+        voice = self.bot.lavalink.player_manager.get(guild)
+        if voice is None:
+            return {}, sc.OK_200
+        else:
+            songs = []
+            for s in voice.queue:
+                songs.append({'title': s.title, 'author': s.author, 'duration': s.duration, 'uri': s.uri})
+            return {'songs': songs}, sc.OK_200
+
+    @fetch_guild
+    async def queue_song(self, guild, uid, song):
+        voice = self.bot.lavalink.player_manager.get(guild)
+
+        song = song.strip('<>')
+        if not url_rx.match(song):
+            song = f'ytsearch:{song}'
+
+        results = await voice.node.get_tracks(song)
+
+        if not results or not results['tracks']:
+            return [], sc.BAD_REQUEST_400
+
+        songs = []
+        if results['loadType'] == 'PLAYLIST_LOADED':
+            tracks = results['tracks']
+
+            for track in tracks:
+                voice.add(requester=uid, track=track)
+                songs.append({'title': track.title, 'author': track.author,
```

Comment:
gross

Reply:
Do you want to comment something constructive?

Reply:
yes, I think you should make a class so that songs have a defined structure